### PR TITLE
refactor(mobile): deprecate getUserAssetsByDeviceId

### PR DESCRIPTION
**Pull request by shenlong-tanwen**
#### Changes made in the PR:

- Deprecated the `/assets/:deviceId` endpoint
- Created a new endpoint `/search/device/:deviceId` which is to be used instead of the deprecated endpoint

### CODE REVIEW

**Review by jrasm91 — COMMENTED** [review]
Looks great my only concern is merge conflicts with putting it in search service. Maybe the other asset controller would be better. What about /asset/device/:id?

**Comment by shenlong-tanwen** [comment]
> Looks great my only concern is merge conflicts with putting it in search service. Maybe the other asset controller would be better. What about /asset/device/:id?

Should be good now. Moved it to asset controller under `/asset/device/:deviceId`

**Review by jrasm91 — APPROVED** [review]
Thanks for the quick turn around!

**Comment by jrasm91** [comment]
That sounds right, the old asset controller doesn't actually use validation for most stuff.

**Comment by shenlong-tanwen** [comment]
> That sounds right, the old asset controller doesn't actually use validation for most stuff.

Welp! 😅 Fixed it now, the `deviceId` from the mobile device is actually a UDID which is just a hex string and not a UUID. Updated the validator to validate the ID to be a non empty string since the web sends a non-hex `deviceId`

**Comment by jrasm91** [comment]
Sounds good!

File: server/src/immich/api-v1/asset/asset.controller.ts [hunk 1]
```diff
@@ -147,9 +147,10 @@ export class AssetController {
   }
 
   /**
-   * Get all asset of a device that are in the database, ID only.
+   * DEPRECATED! Use /asset/device/:deviceId instead
```
Comment (jrasm91) [inline server/src/immich/api-v1/asset/asset.controller.ts:150]: You can actually use `@deprecated` btw

**Review by jrasm91 — APPROVED** [review]

**Comment by alextran1502** [comment]
@shenlong-tanwen FYI, I tested with the new API, and it works now. To prevent breaking changes, I put back the old API. We will remove this and replace it with the new API in 1.92 - 3 releases from now.

**Pull request merged**